substrate-archive/src/tasks.rs

```diff
@@ -166,9 +186,52 @@ where
 			storage_changes: storage_changes.main_storage_changes,
 			child_storage: storage_changes.child_storage_changes,
 			block_hash: hash,
-			block_num: num,
+			block_num: number,
 		})
 	}
+
+	fn execute_with_tracing(self, targets: &str) -> Result<(BlockChanges<Block>, Traces), ArchiveError> {
+		let BlockExecutor { block, backend, id, api } = self;
+		let BlockPrep { block, state, hash, parent_hash, number } = Self::prepare_block(block, &backend, &id)?;
+
+		let span_events = Arc::new(Mutex::new(SpansAndEvents { spans: Vec::new(), events: Vec::new() }));
+		{
+			let handler = TraceHandler::new(&targets, number.into(), hash.as_ref().to_vec(), span_events.clone());
+			let dispatch = Dispatch::new(handler);
+
+			let dispatcher_span = tracing::debug_span!(
+				target: "state_tracing",
+				"execute_block",
+				extrinsics_len = block.extrinsics().len()
+			);
+			let _guard = dispatcher_span.enter();
+			let res = dispatcher::with_default(&dispatch, || {
+				let span = tracing::info_span!("block_trace", "trace_block");
+				let _enter = span.enter();
+				api.execute_block(&id, block)
+			});
+			if res.is_err() {
+				log::error!("{:?}", res);
+			}
+		}
+
+		let changes =
+			api.into_storage_changes(&state, None, parent_hash).map_err(ArchiveError::ConvertStorageChanges)?;
+
+		let changes = BlockChanges {
+			storage_changes: changes.main_storage_changes,
+			child_storage: changes.child_storage_changes,
+			block_hash: hash,
+			block_num: number,
+		};
+		// We destroy the Arc and transform the Mutex here in order to avoid additional allocation.
+		// The Arc is cloned into the thread-local tracing subscriber in the scope of `storage`, creating
```

Comment:
Ah yeah, sorry this is confusing because previously the tracing was collected inside a 

```rust
let storage = {
    // ... block execution code
    return storage_changes;
};
```
but this change removes that. in this case the scope would be this part of the code (it is no longer named)

```rust
{
			let handler = TraceHandler::new(&targets, number.into(), hash.as_ref().to_vec(), span_events.clone());
			let dispatch = Dispatch::new(handler);

			let dispatcher_span = tracing::debug_span!(
				target: "state_tracing",
				"execute_block",
				extrinsics_len = block.extrinsics().len()
			);
			let _guard = dispatcher_span.enter();
			dispatcher::with_default(&dispatch, || {
				let span = tracing::info_span!("block_trace", "trace_block");
				let _enter = span.enter();
				api.execute_block(&id, block)
			})?;
		}

```
We need it to close the `execute_block`span properly and not include anything beyond the block execution


This comment is removed since we no longer unwrap the mutex, though